### ISSUE TITLE
PowerPC support

### ISSUE THREAD

**Issue by pkubaj** [pika-org/pika#339]
## Expected Behavior

Build should finish.

## Actual Behavior

Build fails:
```
/usr/bin/c++ -DNDEBUG -DPIKA_EXPORTS -D_GNU_SOURCE -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/executors/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/executors/include -I/wrkdirs/usr/ports/devel/pika/work/.build -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/async_base/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/async_base/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/allocator_support/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/allocator_support/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/concepts/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/concepts/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/config/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/config/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/preprocessor/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/preprocessor/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/type_support/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/type_support/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/assertion/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/assertion/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/coroutines/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/coroutines/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/errors/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/errors/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/filesystem/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/filesystem/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/format/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/format/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/logging/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/logging/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/timing/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/timing/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/hardware/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/hardware/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/thread_support/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/thread_support/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/functional/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/functional/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/datastructures/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/datastructures/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/debugging/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/debugging/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/itt_notify/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/itt_notify/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/tag_invoke/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/tag_invoke/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/memory/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/memory/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/util/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/util/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/iterator_support/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/iterator_support/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/execution/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/execution/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/async_combinators/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/async_combinators/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/futures/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/futures/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/synchronization/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/synchronization/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/execution_base/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/execution_base/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/lock_registration/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/lock_registration/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/concurrency/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/concurrency/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/hashing/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/hashing/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/threading_base/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/threading_base/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/affinity/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/affinity/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/topology/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/topology/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/pack_traversal/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/pack_traversal/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/threading/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/threading/include -I/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/properties/include -I/wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/properties/include -isystem /usr/local/include -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -fPIC -pthread -Wall -Wextra -Wpedantic -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-noexcept-type -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -std=c++17 -MD -MT libs/pika/executors/CMakeFiles/pika_executors.dir/Unity/unity_0_cxx.cxx.o -MF libs/pika/executors/CMakeFiles/pika_executors.dir/Unity/unity_0_cxx.cxx.o.d -o libs/pika/executors/CMakeFiles/pika_executors.dir/Unity/unity_0_cxx.cxx.o -c /wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/executors/CMakeFiles/pika_executors.dir/Unity/unity_0_cxx.cxx
In file included from /wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/executors/CMakeFiles/pika_executors.dir/Unity/unity_0_cxx.cxx:12:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/executors/src/fork_join_executor.cpp:7:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/executors/include/pika/executors/fork_join_executor.hpp:22:
In file included from /wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/hardware/include/pika/modules/hardware.hpp:15:
/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/hardware/include/pika/hardware/cpuid.hpp:90:2: error: Unsupported platform.
```

## Steps to Reproduce the Problem

Build pika on powerpc64le.

## Specifications

  - pika version 0.6.0
  - Platform FreeBSD 13.1-RELEASE, powerpc64le, LLVM 13.0.0

**Comment by msimberg** [comment]
@pkubaj thanks for the report. Fundamentally the important part of that build failure is
```
error: Unsupported platform.
```
We do not currently support building for PowerPC. That said, I think we don't currently use that header at all and it can be removed. Would you mind trying to build this PR: #341. Note that if you haven't already done so, you'll need to configure pika with `PIKA_WITH_GENERIC_CONTEXT_COROUTINES=ON`.

**Comment by msimberg** [comment]
@pkubaj this was automatically closed by #341 being merged. However, I don't know if it actually fixed your build. Please feel free to reopen if there are remaining build errors.

**Comment by pkubaj** [comment]
Thank you, this indeed fixed the build.
